### PR TITLE
Group/team cache refactoring

### PR DESCRIPTION
**Pull request by cdoremus**
This PR includes
* Moving group/team fetching out of the GroupPage into the containers.
* Works with un-cached or cached team
* A team issue URL works with cached team
* Get team issue URL to work with an un-cached team.
* Cleanup and tests

### CODE REVIEW

File: src/components/groups/GroupPage.tsx [hunk 1]
```diff
@@ -121,42 +104,42 @@ class GroupPage extends React.Component<Props, State> {
...
+            currentGroup={group ? group : undefined}
             issues={this.props.issues}
             issueId={this.props.match.params.issueid}
           >
             <div className="page__group">
               <div className="page__header">
-                <div className="page__header__image"><img alt={group.name} src={groupImage}/></div>
-                <h1 className="page__title">{group.name}</h1>
-                <h2 className="page__subtitle">{group.subtitle}&nbsp;</h2>
+                <div className="page__header__image"><img alt={group ? group.name : ''} src={groupImage}/></div>
```
Comment (nickoneill) [inline src/components/groups/GroupPage.tsx]: We have a group of these ternaries in this block, perhaps we can move it above this block and do it once.

Comment (cdoremus) [reply src/components/groups/GroupPage.tsx]: Yea, that's the kind of cleanup I'd like to do once I get the team issue URL to work with an un-cached team.
Please suggest any other cleanup items you see.

**Comment by cdoremus** [comment]
Just pushed 2 commits that fixed issues with calling a group issue URL with un-cached group data. Also removed extra group API calls that were being made. Please test this functionality. In the meantime, I'll start on the cleanup work.

**Comment by nickoneill** [comment]
The "additionalComponents" stuff looks really interesting, I didn't know you could do that.

**Review by nickoneill — APPROVED** [review]

**Comment by nickoneill** [comment]
Looks awesome, are you done with the cleanup work you were doing?

**Pull request merged**